Source/ACE.DatLoader/FileTypes/CharGen.cs

```diff
@@ -0,0 +1,228 @@
+﻿using ACE.DatLoader.Entity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ACE.DatLoader.FileTypes
+{
+    public class CharGen
+    {
+        public int Did { get; set; }
+        public List<List<Loc>> StarterAreas { get; set; } = new List<List<Loc>>();
+        public Dictionary<int, HeritageGroupCG> HeritageGroups { get; set; } = new Dictionary<int, HeritageGroupCG>();
+
+        public static CharGen ReadFromDat(string datFilePath, uint offset, uint size, uint sectorSize)

```

Comment:
Is CharGen essentially a singleton?  That is, there's only ever one of them?  If so, just leave this as a todo:
// TODO: refactor into a singleton pattern